pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,86 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers.testing import check_figures_equal
+
+
+@check_figures_equal()
+def test_solar_set_terminator_datetime_string():
+    """
+    Test passing the solar argument with the civil terminator and a datetime
+    string.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.solar(R="d", J="W0/15c", B="a", T="c+d1990-02-17T04:25:00")
+    fig_test.solar(
+        R="d",
+        J="W0/15c",
+        B="a",
+        terminator="civil",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_solar_default_terminator():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.solar(R="d", J="W0/15c", B="a", T="d+d1990-02-17T04:25:00")
+    fig_test.solar(R="d", J="W0/15c", B="a", terminator_datetime="1990-02-17 04:25:00")
+    return fig_ref, fig_test
```

Comment:
Once we finish off the dvc branch at #1036, we should refactor this to use the simpler `@pytest.mark.mpl_image_compare`. See also specific comments at https://github.com/GenericMappingTools/pygmt/pull/1036#discussion_r595519351.

Appreciate your patience on this while we streamline our workflow :grin: